drivers/gpio/gpio_rpi_pico.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2021, Yonatan Schachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <drivers/gpio.h>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
```

Comment:
why are these needed?

Reply:
Some header files high in the dependency tree define static, always-inline functions. These are not used by every source file, but are included anyway, causing many unused-function warnings

Reply:
Other HALs like STM32 LL API also has lots of `static inline` functions and doesn't require this.

Reply:
I'm not sure how this works for the ST. Perhaps all those functions are always used? Anyway, on the rpi it causes dozens of warnings, so as far as I know it's either turning this warning off for the whole file or using diagnostic push to ignore it only on the HAL provided functions

Reply:
> I'm not sure how this works for the ST. Perhaps all those functions are always used? Anyway, on the rpi it causes dozens of warnings, so as far as I know it's either turning this warning off for the whole file or using diagnostic push to ignore it only on the HAL provided functions

Most of the functions are not used in STM32 LL. I suspect there's something wrong here, these pragmas should not be needed.

Reply:
You are right. I removed the pragmas and it worked fine. Not sure what caused this initially, but I can't reproduce it now.